### PR TITLE
drivers: sensor: mcp9600: Added driver and sample for mcp9600

### PR DESCRIPTION
**Pull request by riedlse**
Added driver and sample for mcp9600 microchip I2C thermocouple
amplifier. Tested on frdm_kw41z and nrf52850dk.

Signed-off-by: Steven Riedl <steve@serconsultingllc.com>

### CODE REVIEW

File: drivers/sensor/mcp9600/mcp9600.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	int size = 6;
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, size);
+	if (ret < 0) {
+		return ret;
+	}
+	data->ttemp = (buf[0]<<8) | (buf[1]);
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.c:54]: I believe spaces around the `<<` operator is more standard, though that might not be present in all existing code.  The array reference operand `buf[1]` doesn't need parentheses around it; the precedence should be clear enough for anybody even if they don't remember whether `<<` or `|` is higher.

File: drivers/sensor/mcp9600/mcp9600.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+static int mcp9600_channel_get(const struct device *dev,
+			enum sensor_channel chan,
+			struct sensor_value *val)
+{
+	struct mcp9600_data *data = dev->data;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+			temp = data->ttemp>>4;
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.c:70]: Indentation here looks wrong: two tabs to statement?

File: drivers/sensor/mcp9600/mcp9600.h [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#define MCP9600_CFG_ALERT_MODE_INT	BIT(1)
+#define MCP9600_CFG_ALERT_HI_LO		BIT(2)
+#define MCP9600_CFG_ALERT_RIS_FAL	BIT(3)
+#define MCP9600_CFG_ALERT_TH_TC		BIT(4)
+#define MCP9600_CFG_INT_CLEAR		BIT(7)
+
+struct mcp9600_data {
+	const struct device *i2c_master;
+
+	int16_t ttemp;
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.h:59]: This ordering wastes memory due to padding between the `int16_t` and `uint32_t` members.  Probably best to group them by whole and fraction.  Or simply store the value in an `int32_t` in some units that cover the sensor range and resolution, e.g. mCel, or what appears to be the native resolution: Cel/16.

Otherwise please document the relationship between the whole and fractional values with respect to sign and units.  I.e. is the true temperature `dtemp + dfrac / 1000000`?

File: drivers/sensor/mcp9600/mcp9600_trigger.c [hunk 4]
```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2020 SER Consulting LLC, Steven Riedl
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ *
+ * !!!! NOT DONE OR TESTED, Converted from mcp9808
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600_trigger.c:5]: This does not inspire confidence....

File: samples/sensor/mcp9600/README.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
+.. _mcp9600:
+
+mcp9600 Temperature Sensor
+###################################
+
+Overview
+********
+
+This sample application periodically reads temperature data from the first
```
Comment (pabigot) [inline samples/sensor/mcp9600/README.rst:9]: Please add a link to manufacturer information about this sensor.

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2020, SER Consulting LLC  / Steven Riedl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c0 {
+	mcp9600@60 {
+		compatible = "microchip,mcp9600";
```
Comment (pabigot) [inline samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:9]: Need the binding `dts/bindings/sensor/microchip,mcp9600`.

File: samples/sensor/mcp9600/src/main.c [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+			} else {
+				struct sensor_value ttemp;
+
+				sensor_channel_get(mcp,
+					SENSOR_CHAN_AMBIENT_TEMP, &ttemp);
+#ifdef CONFIG_NEWLIB_LIBC_FLOAT_PRINTF
+				printf("Temperature: %.4f C\n",
+					sensor_value_to_double(&ttemp));
+#else
+				printk("Temperature: %d\n", ttemp.val1);
```
Comment (pabigot) [inline samples/sensor/mcp9600/src/main.c:57]: Include units in the output for non-floating point.

**Review by pabigot — CHANGES_REQUESTED** [review]
Please also squash the commits and update the description to cover the content of the PR.

File: drivers/sensor/mcp9600/mcp9600.c (hunk 1 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:54]: ok, check patch seems not to care, but I can do that.

File: drivers/sensor/mcp9600/mcp9600.c (hunk 2 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:70]: will check

File: drivers/sensor/mcp9600/mcp9600.c [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+	.i2c_bus = DT_INST_BUS_LABEL(0),
+	.i2c_addr = DT_INST_REG_ADDR(0),
+#ifdef CONFIG_MCP9600_TRIGGER
+	.alert_pin = DT_INST_GPIO_PIN(0, int_gpios),
+	.alert_flags = DT_INST_GPIO_FLAGS(0, int_gpios),
+	.alert_controller = DT_INST_GPIO_LABEL(0, int_gpios),
+#endif /* CONFIG_MCP9600_TRIGGER */
+};
+
+DEVICE_AND_API_INIT(mcp9600, DT_INST_LABEL(0), mcp9600_init,
```
[the remark this answers is not in the document]
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:137]: OK

File: drivers/sensor/mcp9600/mcp9600_trigger.c (hunk 4 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600_trigger.c:5]: Missed that, was actually well tested, even ran some fake value insertion tests to insure the negative temps were correct, will remove...

File: samples/sensor/mcp9600/README.rst (hunk 5 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst:9]: ok

File: samples/sensor/mcp9600/src/main.c (hunk 7 above)
Comment (riedlse) [reply samples/sensor/mcp9600/src/main.c:57]: ok

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay (hunk 6 above)
Comment (riedlse) [reply samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:9]: need to figure this one out

File: drivers/sensor/mcp9600/mcp9600.h (hunk 3 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.h:59]: Actually am not using the other variables, so they were a waste anyway, removed

File: drivers/sensor/mcp9600/mcp9600_trigger.c (hunk 4 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600_trigger.c:5]: Actually trigger was not tested, just removed all trigger functionality. Can be added back in at a later time properly as the device has 4 triggers.

File: drivers/sensor/mcp9600/mcp9600.c (hunk 8 above)
[the remark this answers is not in the document]
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:137]: fixed

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay (hunk 6 above)
Comment (riedlse) [reply samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:9]: Never knew about that... Found and added

File: samples/sensor/mcp9600/src/main.c (hunk 7 above)
Comment (riedlse) [reply samples/sensor/mcp9600/src/main.c:57]: added

File: samples/sensor/mcp9600/README.rst (hunk 5 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst:9]: added

File: drivers/sensor/CMakeLists.txt [hunk 9]
```diff
@@ -36,7 +36,7 @@ add_subdirectory_ifdef(CONFIG_ISL29035		isl29035)
 add_subdirectory_ifdef(CONFIG_ISM330DHCX	ism330dhcx)
 add_subdirectory_ifdef(CONFIG_LIS2DH		lis2dh)
 add_subdirectory_ifdef(CONFIG_LIS2DS12		lis2ds12)
-add_subdirectory_ifdef(CONFIG_LIS2DW12          lis2dw12)
+add_subdirectory_ifdef(CONFIG_LIS2DW12		lis2dw12)
```
Comment (MaureenHelm) [inline drivers/sensor/CMakeLists.txt:39]: Please move unrelated whitespace changes to a separate commit or remove from this PR

File: drivers/sensor/CMakeLists.txt [hunk 10]
```diff
@@ -67,7 +68,7 @@ add_subdirectory_ifdef(CONFIG_SI7060		si7060)
 add_subdirectory_ifdef(CONFIG_SM351LT		sm351lt)
 add_subdirectory_ifdef(CONFIG_STTS751		stts751)
 add_subdirectory_ifdef(CONFIG_SX9500		sx9500)
-add_subdirectory_ifdef(CONFIG_TH02		    th02)
+add_subdirectory_ifdef(CONFIG_TH02			th02)
```
Comment (MaureenHelm) [inline drivers/sensor/CMakeLists.txt:71]: Another unrelated whitespace change

File: drivers/sensor/mcp9600/mcp9600.c [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+#include <drivers/i2c.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include "mcp9600.h"
+
+#define DT_DRV_COMPAT microchip_mcp9600
+
+#define MCP9600_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
```
Comment (MaureenHelm) [inline drivers/sensor/mcp9600/mcp9600.c:20]: Does this sensor support SPI too? I don't see this macro being used anywhere

File: drivers/sensor/mcp9600/mcp9600.c [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+static int mcp9600_channel_get(const struct device *dev,
+			enum sensor_channel chan,
+			struct sensor_value *val)
+{
+	struct mcp9600_data *data = dev->data;
+	int32_t temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temp = (int32_t)(int16_t)data->ttemp;
```
Comment (MaureenHelm) [inline drivers/sensor/mcp9600/mcp9600.c:66]: why the double cast?

File: samples/sensor/mcp9600/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(lm73c)
```
Comment (MaureenHelm) [inline samples/sensor/mcp9600/CMakeLists.txt:5]: Rename the project to mcp9600

File: samples/sensor/mcp9600/README.rst [hunk 14]
```diff
@@ -0,0 +1,49 @@
+.. _mcp9600:
+
+mcp9600 Temperature Sensor
```
Comment (MaureenHelm) [inline samples/sensor/mcp9600/README.rst:3]: Please split the driver and sample application into two commits

File: drivers/sensor/mcp9600/Kconfig [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+	int "Sensor delayed work thread stack size"
+	depends on MCP9600_TRIGGER_OWN_THREAD
+	default 1024
+
+config MCP9600_THREAD_PRIORITY
+	int "MCP9600 thread priority"
+	depends on MCP9600_TRIGGER_OWN_THREAD
+	default 10
+
+endif # MCP9600
```
Comment (MaureenHelm) [inline drivers/sensor/mcp9600/Kconfig:47]: This can all be removed since you removed the trigger support

File: drivers/sensor/mcp9600/CMakeLists.txt [hunk 16]
```diff
@@ -0,0 +1,6 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_MCP9600 mcp9600.c)
+zephyr_library_sources_ifdef(CONFIG_MCP9600_TRIGGER mcp9600_trigger.c)
```
Comment (MaureenHelm) [inline drivers/sensor/mcp9600/CMakeLists.txt:6]: This line can be removed

File: samples/sensor/mcp9600/sample.yaml [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+  sample.sensor.mcp9600:
+    harness: sensor
+    tags: sensors
+    harness_config:
+        type: one_line
+        regex:
+            - "ttemp: (.*)"
+        fixture: fixture_i2c_mcp9600
+    depends_on: i2c
+    filter: dt_compat_enabled("microchip,mcp9600")
```
Comment (MaureenHelm) [inline samples/sensor/mcp9600/sample.yaml:13]: I don't think this will pick up the boards you added overlays for, so please replace this with a platform_allow list

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: drivers/sensor/mcp9600/mcp9600.c (hunk 11 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:20]: I2C only, will check why the macro is still there, probably not needed.

File: drivers/sensor/CMakeLists.txt (hunk 9 above)
Comment (riedlse) [reply drivers/sensor/CMakeLists.txt:39]: Will fix on new PR

File: drivers/sensor/CMakeLists.txt (hunk 10 above)
Comment (riedlse) [reply drivers/sensor/CMakeLists.txt:71]: Will fix on split PR

File: drivers/sensor/mcp9600/mcp9600.c (hunk 12 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:66]: Peter Bigot pointed that out easy way to get it to be sign extended from an uint16_t to int32_t, thought it was kind of elegant. I'll add a comment

Comment (pabigot) [reply drivers/sensor/mcp9600/mcp9600.c:66]: That'd be from me: the `(int16_t)` is necessary to convert from the unsigned `uint16_t` because the underlying value is signed.  The outer `(int32_t)` I left in my recommendation because IIRC MISRA doesn't like implicit type conversions.

I'm fine with removing the outer one; absent MISRA I wouldn't have it there myself.

File: samples/sensor/mcp9600/CMakeLists.txt (hunk 13 above)
Comment (riedlse) [reply samples/sensor/mcp9600/CMakeLists.txt:5]: fixed, will do a new PR separate and close this one.

File: samples/sensor/mcp9600/README.rst (hunk 14 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst:3]: OK easier for me to close this one and create two new ones.

File: samples/sensor/mcp9600/CMakeLists.txt (hunk 13 above)
Comment (pabigot) [reply samples/sensor/mcp9600/CMakeLists.txt:5]: Why?

Comment (pabigot) [reply samples/sensor/mcp9600/CMakeLists.txt:5]: Just split the commits locally, and force-push an updated branch:

    git push r.whatever +sensor_mcp9600

File: drivers/sensor/mcp9600/Kconfig (hunk 15 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/Kconfig:47]: Correct, Peter said to move it to dts as well, if I add it back in ever...

File: drivers/sensor/mcp9600/CMakeLists.txt (hunk 16 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/CMakeLists.txt:6]: done

File: drivers/sensor/mcp9600/mcp9600.c (hunk 11 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:20]: done on next PR

File: samples/sensor/mcp9600/sample.yaml (hunk 17 above)
Comment (riedlse) [reply samples/sensor/mcp9600/sample.yaml:13]: Switched to the MCP9808 sample.yaml, hope that is acceptable as Peter mentioned not sure what boards/chips are supported.

File: samples/sensor/mcp9600/CMakeLists.txt (hunk 13 above)
Comment (riedlse) [reply samples/sensor/mcp9600/CMakeLists.txt:5]: The separate PR is for the request below to split the driver and sample in to two commits, but my git knowledge may be lacking (use it a lot, but this is my first time adding to a large project like this)

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay [hunk 18]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2020, SER Consulting LLC  / Steven Riedl
```
Comment (pabigot) [inline samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:2]: Have you actually tested on this?  It's an unusual board to use for external connections.

If you don't need it, it might be better to drop this and just leave the DK variant of the same SOC.

File: samples/sensor/mcp9600/README.rst [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+        Temperature: 20.3125 C
+        Temperature: 20.5000 C
+        Temperature: 20.7500 C
+        Temperature: 20.8125 C
+        Temperature: 20.6875 C
+        Temperature: 20.8125 C
+        Temperature: 20.8750 C
+        Temperature: 20.6250 C
+
+.. _`MCP9600 Datasheet`: https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BME280-DS002.pdf
```
Comment (pabigot) [inline samples/sensor/mcp9600/README.rst]: No, this is the BME280 data sheet.    Better to link to https://www.microchip.com/wwwproducts/en/MCP9600 which allows people to find whatever they want about this sensor, including but not limited to the datasheet.

File: samples/sensor/mcp9600/README.rst [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+   :compact:
+
+Sample Output
+=============
+To check output of this sample , any serial console program can be used.
+This example uses screen on OS X with a frdm_kw41z development board.
+
+.. code-block:: console
+
+        $ screen /dev/tty.usbmodem0006210000001 115200
```
Comment (pabigot) [inline samples/sensor/mcp9600/README.rst:34]: That seems unlikely to be what anybody else would use.  Maybe `/dev/ttyACM0` like most other examples that suggest using screen?

File: samples/sensor/mcp9600/prj.conf [hunk 21]
```diff
@@ -0,0 +1,9 @@
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_I2C=y
+CONFIG_GPIO=y
+CONFIG_SENSOR=y
+CONFIG_MCP9600=y
+CONFIG_BUILD_OUTPUT_HEX=y
+CONFIG_NEWLIB_LIBC=y
```
Comment (pabigot) [inline samples/sensor/mcp9600/prj.conf:7]: You shouldn't need newlib now the Zephyr natively knows how to format floating point.  You do need  CBPRINTF_FP_SUPPORT.

File: samples/sensor/mcp9600/prj.conf [hunk 22]
```diff
@@ -0,0 +1,9 @@
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_I2C=y
+CONFIG_GPIO=y
+CONFIG_SENSOR=y
+CONFIG_MCP9600=y
+CONFIG_BUILD_OUTPUT_HEX=y
```
Comment (pabigot) [inline samples/sensor/mcp9600/prj.conf:6]: Should not need to specify the build output.

File: samples/sensor/mcp9600/src/main.c [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+			ret = sensor_sample_fetch(mcp);
+			if (ret < 0) {
+				printk("\"%s\" fetch error %d\n",
+					   MCP9600_LABEL, ret);
+			} else {
+				struct sensor_value ttemp;
+
+				sensor_channel_get(mcp,
+					SENSOR_CHAN_AMBIENT_TEMP, &ttemp);
+#ifdef CONFIG_NEWLIB_LIBC_FLOAT_PRINTF
```
Comment (pabigot) [inline samples/sensor/mcp9600/src/main.c:53]: You're selecting support for floating point output in Kconfig, so you don't need to make it conditional in here.

I'm suspecting there's some samples you used as a template that need to get cleaned up....

File: drivers/sensor/mcp9600/mcp9600.c [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+			enum sensor_channel chan)
+{
+	struct mcp9600_data *data = dev->data;
+	uint8_t buf[2];
+	int size = 2;
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, size);
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.c:48]: Replace `size` with `sizeof(buf)`.

File: drivers/sensor/mcp9600/mcp9600.c [hunk 25]
```diff
@@ -0,0 +1,119 @@
...
+	if (ret < 0) {
+		return ret;
+	}
+	data->ttemp = (buf[0] << 8) | buf[1];
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.c:52]: If you make this:

    data->ttemp = (int16_t)((buf[0] << 8) | buf[1]);

and declare `ttemp` as signed you get rid of the cast when the value is extracted, and store the value in format closer to ready-to-use (Cel/16).  You could do `(int16_t)(uint16_t)((buf[0] << 8) | buf[1])` if necessary (shouldn't be, except again for safety-related code guidelines).

File: drivers/sensor/mcp9600/mcp9600.c [hunk 26]
```diff
@@ -0,0 +1,119 @@
...
+	.sample_fetch = mcp9600_sample_fetch,
+	.channel_get = mcp9600_channel_get,
+};
+
+static int mcp9600_chip_init(const struct device *dev)
+{
+	uint8_t buf[2];
+
+	int rc = mcp9600_reg_read(dev, MCP9600_REG_ID_REVISION,
+			buf, 2);
```
Comment (pabigot) [inline drivers/sensor/mcp9600/mcp9600.c:88]: s/2/sizeof(buf)/.

**Review by pabigot — CHANGES_REQUESTED** [review]

**Comment by pabigot** [comment]
Also: squash the second commit which fixes a whitespace issue diagnosed in the first commit.  Unless @MaureenHelm wants the sample in a separate commit from the driver as with #31064; that's not an expectation I have.

Re https://github.com/zephyrproject-rtos/zephyr/pull/31064#discussion_r552972211
> OK, will pull this one and do separate, I guess driver has to go first and be merged?

No, put both commits in the same PR, driver first, then sample.

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay (hunk 18 above)
Comment (riedlse) [reply samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:2]: Yes, actually on this (of course you have to wire up an I2C port ) I actually have a custom board that I also use, but makes sense to remove this, mainly tested on the kw and nrfdk.

File: samples/sensor/mcp9600/README.rst (hunk 19 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst]: ARGGGGGG, could have sworn that was in there, trying to be careful with the edits and must have missed that one.

File: samples/sensor/mcp9600/README.rst (hunk 20 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst:34]: Not sure it makes that much of a difference? I do think I say I tested it on a Mac and that's what you get.

File: samples/sensor/mcp9600/prj.conf (hunk 21 above)
Comment (riedlse) [reply samples/sensor/mcp9600/prj.conf:7]: I will test it without that.

File: samples/sensor/mcp9600/prj.conf (hunk 22 above)
Comment (riedlse) [reply samples/sensor/mcp9600/prj.conf:6]: Actually for the KW41Z the flash seems to prefer BIN although will work with Hex, so I put that in there to eliminate a flash warning. Guess it could disappear.

File: samples/sensor/mcp9600/src/main.c (hunk 23 above)
Comment (riedlse) [reply samples/sensor/mcp9600/src/main.c:53]: That would be correct. should be conditional on CBPRINTF_FP_SUPPORT ?

File: drivers/sensor/mcp9600/mcp9600.c (hunk 25 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:52]: Never trusted compilers to not sign extend when they shouldn't and to not do it when they should. I prefer any of the more explicit methods. I think there is a micro in zephyr to do the byte swap as well. Would have to go through the other drivers to find it again.

File: drivers/sensor/mcp9600/mcp9600.c (hunk 26 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:88]: In this case, yes.

File: samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay (hunk 18 above)
Comment (riedlse) [reply samples/sensor/mcp9600/boards/nrf52840dongle_nrf52840.overlay:2]: Deleted, will be in next push.

File: samples/sensor/mcp9600/README.rst (hunk 19 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst]: next commit, done

File: samples/sensor/mcp9600/README.rst (hunk 20 above)
Comment (riedlse) [reply samples/sensor/mcp9600/README.rst:34]: added additional text.

File: drivers/sensor/mcp9600/mcp9600.c (hunk 25 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:52]: actually made it a int32, moved the sign change and multiply up so I could use a value of 1 to invalidate the value on a read failure. There actually is a safety component to my use of this device.

Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:52]: tested

File: drivers/sensor/mcp9600/mcp9600.c (hunk 24 above)
Comment (riedlse) [reply drivers/sensor/mcp9600/mcp9600.c:48]: fixed

File: samples/sensor/mcp9600/src/main.c (hunk 23 above)
Comment (riedlse) [reply samples/sensor/mcp9600/src/main.c:53]: CBPRINTF works removed newlib

File: samples/sensor/mcp9600/prj.conf (hunk 21 above)
Comment (riedlse) [reply samples/sensor/mcp9600/prj.conf:7]: done, tested

File: samples/sensor/mcp9600/CMakeLists.txt (hunk 13 above)
Comment (MaureenHelm) [reply samples/sensor/mcp9600/CMakeLists.txt]: See the [Contribution Workflow](https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow) for some git tips.

**Comment by riedlse** [comment]
Don't have time to finish this right now, will probably resubmit later, with proper multi-device support

**Comment by riedlse** [comment]
wanted to close on last comment
